### PR TITLE
Updated codelab: Write a Flutter desktop application Codelab (github package version)

### PR DESCRIPTION
**Pull request by miquelbeltran**
@domesticmouse @RedBrogdon 

This is the matching PR to the changes in the document "Write a Flutter desktop application Codelab (github package version)"

Some notes:

- I moved the codelab project from `github-graphql-client` to `github-client`.
- All steps are created from scratch using the latest flutter beta, to ensure being as recent as possible.
- Screenshots in the document has been updated to match the app. There should not be any reference to graphql anymore.
- Although the GitHub Dart package provides its own OAuth2 auth logic, I kept the original one and used the resulting access token instead to configure the GitHub Dart package.

## Pre-launch Checklist

- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by miquelbeltran** [comment]
I realized that the checks fail because I used flutter beta and the CI expect an SDK version or 2.15.1 max

**Comment by miquelbeltran** [comment]
Note that all the `−547,691` changes are due to the graphql generated code being removed :)

**Comment by domesticmouse** [comment]
Can you please change https://github.com/miquelbeltran/codelabs/blob/mb-github-client/flutter_ci_script_beta.sh#L15 and https://github.com/miquelbeltran/codelabs/blob/mb-github-client/flutter_ci_script_stable.sh#L15 to reflect the name change of the codelab?

**Comment by miquelbeltran** [comment]
done!

**Comment by domesticmouse** [comment]
Looks like the CI isn't happy:

```console
== Testing 'github-client/window_to_front'
Analyzing window_to_front...

  error - test/window_to_front_test.dart:21:32 - The getter 'platformVersion' isn't defined for the type 'WindowToFront'. Try importing the library that defines 'platformVersion', correcting the name to the name of an existing getter, or defining a getter or field named 'platformVersion'. - undefined_getter
   info - test/window_to_front_test.dart:11:39 - Avoid annotating types for function expression parameters. - avoid_types_on_closure_parameters
```

**Comment by miquelbeltran** [comment]
I removed the test template from the window_to_front project

**Comment by domesticmouse** [comment]
I realise this is going to be a PITA, but can we split the adding of the new codelab and deleting the old codelab into two separate PRs? I'm going to need to play an interesting game of sequencing to get this change to land right.

(There are links in the codelab that point at incorrect URLs, but I need to land the new codelab to make sure the fixups are correct, before deleting the old codelab and old code)

**Comment by domesticmouse** [comment]
Also, you've updated the Copyright lines, even though in some cases it was only a move. Please revert

**Comment by domesticmouse** [comment]
Actually, the list of changes for macOS was kinda silly, so I've pushed a cleanup version to this PR.

File: github-client/step_06/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# github_client
```
Comment (domesticmouse) [inline github-client/step_06/README.md:1]: plz delete

File: github-client/step_06/lib/github_oauth_credentials.dart [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2020 Google LLC
+// Copyright 2022 Google LLC
```
Comment (domesticmouse) [inline github-client/step_06/lib/github_oauth_credentials.dart:1]: please revert this date change

File: github-client/step_06/lib/main.dart [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2020 Google LLC
+// Copyright 2022 Google LLC
```
Comment (domesticmouse) [inline github-client/step_06/lib/main.dart:1]: please revert

File: github-client/step_07/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# github_client
```
Comment (domesticmouse) [inline github-client/step_07/README.md:1]: please delete this file

File: github-client/step_07/lib/github_oauth_credentials.dart [hunk 5]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2020 Google LLC
+// Copyright 2022 Google LLC
```
Comment (domesticmouse) [inline github-client/step_07/lib/github_oauth_credentials.dart:1]: please revert this change

File: github-client/step_07/lib/main.dart [hunk 6]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2020 Google LLC
+// Copyright 2022 Google LLC
```
Comment (domesticmouse) [inline github-client/step_07/lib/main.dart:1]: please revert this change

**Review by domesticmouse — CHANGES_REQUESTED** [review]
A couple of change required

**Comment by miquelbeltran** [comment]
Going to take care of those asap

**Comment by domesticmouse** [comment]
Thanks!

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
Thanks @miquelbeltran

**Pull request merged**